### PR TITLE
Shp 02

### PR DESCRIPTION
**Pull request by diego-schurch**
- Added support for package size to the application.
- Also, update the frontend to the newest version.

### CODE REVIEW

File: src/main/java/com/shipping/application/controller/PackageSizeController.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import java.util.List;
+
+@RestController
+public class PackageSizeController {
+
+    @Autowired
+    private PackageSizeService packageSizeService;
+
+    @GetMapping(value = "/size/{packageSize}", produces= {MediaType.APPLICATION_JSON_VALUE})
+    public ResponseEntity getAllPackageSizes(@PathVariable String packageSize) throws JsonProcessingException {
```
Comment (grunger94) [inline src/main/java/com/shipping/application/controller/PackageSizeController.java:24]: Do you return the exception to the UI without no previous processing like logging the actual exception and returning a friendly message to the client?

File: src/main/java/com/shipping/application/controller/PackageSizeController.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+        List<String> packageSizes = this.packageSizeService.getPackageSizes(packageSize);
```
Comment (grunger94) [inline src/main/java/com/shipping/application/controller/PackageSizeController.java:26]: Nit: this is not needed here

File: src/main/java/com/shipping/application/repository/PackageSizeRepository.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+@Repository
+public class PackageSizeRepository {
+
+    @Autowired
+    RabbitMQSender sender;
```
Comment (grunger94) [inline src/main/java/com/shipping/application/repository/PackageSizeRepository.java:19]: Private scope

File: src/main/java/com/shipping/application/repository/PackageSizeRepository.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+    private ObjectMapper mapper = new ObjectMapper();
+
+    public List<PackageSizeModel> getPackageSizes(String packageType) throws JsonProcessingException {
+
+        String message = MessageSerializer.requestPackageSizeMessageToPlainTextJson(new RequestPackageSizeMessage("packageSizeByType", packageType));
+        String receivedMessage = sender.sendRequest(message);
+
+        List<PackageSizeModel> packageSizes = mapper.readValue(receivedMessage, new TypeReference<List<PackageSizeModel>>(){});
```
Comment (grunger94) [inline src/main/java/com/shipping/application/repository/PackageSizeRepository.java:28]: You don't need to declare packageSizes, just return the result of mapper.readValue() immediately

File: src/main/java/com/shipping/application/service/impl/PackageSizeServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+    @Autowired
+    private PackageSizeDao packageSizeDao;
+
+    @Override
+    public List<String> getPackageSizes(String packageType) throws JsonProcessingException {
+
+        ArrayList<String> packageSizes = new ArrayList<>();
+
+        for (PackageSizeModel packageSize: this.packageSizeDao.getPackageSizes(packageType)) {
```
Comment (grunger94) [inline src/main/java/com/shipping/application/service/impl/PackageSizeServiceImpl.java:26]: Nit: investigate how to to this with java 8 streams and collectors

File: src/main/java/com/shipping/application/util/MessageSerializer.java [hunk 6]
```diff
@@ -6,4 +6,8 @@ static public String requestMessageToPlainTextJson(RequestMessage requestMessage
             return String.format("{\"type\":\"%s\"}", requestMessage.type);
     }
 
+    static public String requestPackageSizeMessageToPlainTextJson(RequestPackageSizeMessage requestMessage){
```
Comment (grunger94) [inline src/main/java/com/shipping/application/util/MessageSerializer.java:9]: The convention is `public static`

File: src/main/java/com/shipping/application/dao/PackageSizeDao.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.shipping.application.model.PackageSizeModel;
+import com.shipping.application.repository.PackageSizeRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+@Component
+public class PackageSizeDao {
```
Comment (grunger94) [inline src/main/java/com/shipping/application/dao/PackageSizeDao.java:12]: Why do you need a DAO layer? Can't you have your service ask for the data to the repository directly?

File: src/main/java/com/shipping/application/controller/PackageSizeController.java (hunk 1 above)
Comment (diego-schurch) [reply src/main/java/com/shipping/application/controller/PackageSizeController.java:24]: On error returns a 500 response

**Review by Joxebus — APPROVED** [review]

**Pull request merged**